agent/tests/test_events.py

```
@@ -0,0 +1,629 @@
+                func = node.func
+                name = func.attr if isinstance(func, ast.Attribute) else (
+                    func.id if isinstance(func, ast.Name) else None
+                )
+                if name in ("emit", "_on_event") and node.args:
+                    first = node.args[0]
+                    if isinstance(first, ast.Constant) and isinstance(first.value, str):
+                        found.setdefault(first.value, set()).add(path.name)
+                    else:
+                        dynamic.append(f"{path.name}:{first.lineno}")
+            elif isinstance(node, ast.Dict):
+                for key, value in zip(node.keys, node.values):
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)